v2/snmp/session.go

```diff
@@ -0,0 +1,329 @@
+package snmp
+
+import (
+	"context"
+	"encoding/asn1"
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+
+	"github.com/geoffgarside/ber"
+)
+
+// Session provides an interface for SNMP device management.
+type Session interface {
+	// Issues an SNMP GET request for the specified oids.
+	// Get request processing is described at https://tools.ietf.org/html/rfc1905#section-4.2.1.
+	Get(ctx context.Context, oids []string) (*PDU, error)
+
+	// Issues an SNMP GET NEXT request for the specified oids.
+	// Get Bext request processing is described athttps://tools.ietf.org/html/rfc1905#section-4.2.2.
+	GetNext(ctx context.Context, oids []string) (*PDU, error)
+
+	// Issues an SNMP GET BULK request for the specified oids.
+	// Get Bulk request processing is described at https://tools.ietf.org/html/rfc1905#section-4.2.3
+	GetBulk(ctx context.Context, oids []string, nonRepeaters int, maxRepetitions int) (*PDU, error)
+
+	// Issues SNMP GET NEXT requests starting from the specified root oid, invoking the function walker for each
+	// variable that is a descendant of the root oid.
+	Walk(ctx context.Context, rootOid string, walker Walker) error
+
+	// Issues SNMP GET BULK requests starting from the specified root oid, invoking the function walker for each
+	// variable that is a descendant of the root oid.
+	BulkWalk(ctx context.Context, rootOid string, maxRepetitions int, walker Walker) error
```

Comment:
param order, above is ctx, oid, walker, should this be ctx, void, walker maxreps

Reply:
I took the view that maxReps was more closely related to the oid.